server/src/main/java/org/apache/iotdb/db/query/externalsort/MultiSourceExternalSortJobPart.java

```diff
@@ -0,0 +1,61 @@
+ /**
+ * Copyright © 2019 Apache IoTDB(incubating) (dev@iotdb.apache.org)
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ package org.apache.iotdb.db.query.externalsort;
+
+ import java.io.IOException;
+ import java.util.ArrayList;
+ import java.util.List;
+ import org.apache.iotdb.db.query.reader.IPointReader;
+
+
+ public class MultiSourceExternalSortJobPart extends ExternalSortJobPart {
+
+   private String tmpFilePath;
+   private List<ExternalSortJobPart> source;
+   private long queryId;
+
+   public MultiSourceExternalSortJobPart(long queryId, String tmpFilePath,
+       List<ExternalSortJobPart> source) {
+     super(ExternalSortJobPartType.MULTIPLE_SOURCE);
+     this.source = source;
+     this.tmpFilePath = tmpFilePath;
+     this.queryId = queryId;
+   }
+
+   public MultiSourceExternalSortJobPart(long queryId, String tmpFilePath,
+       ExternalSortJobPart... externalSortJobParts) {
+     super(ExternalSortJobPartType.MULTIPLE_SOURCE);
+     source = new ArrayList<>();
+     for (ExternalSortJobPart externalSortJobPart : externalSortJobParts) {
+       source.add(externalSortJobPart);
+     }
+     this.tmpFilePath = tmpFilePath;
+   }
+
+   @Override
+   public IPointReader executeWithGlobalTimeFilter() throws IOException {
+     List<IPointReader> prioritySeriesReaders = new ArrayList<>();
+     for (ExternalSortJobPart part : source) {
+       prioritySeriesReaders.add(part.executeWithGlobalTimeFilter());
```

Comment:
I think `executeForIPointReader` is better than `executeWithGlobalTimeFilter`.

Reply:
fixed